src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigForm.tsx

```diff
@@ -0,0 +1,354 @@
+              mergeMap((templates) => templates),
+              filter((template) => template.provider !== 'Cryostat' || template.name !== 'Cryostat'),
+              toArray(),
+              first()
+            )
+          )
+        )
+        .subscribe(setTemplates)
+    );
+  }, [addSubscription, context.target, context.api, setTemplates]);
+
+  const handleMaxAgeChange = React.useCallback(
+    (evt) => {
+      setMaxAge(Number(evt));
+    },
+    [setMaxAge]
+  );
+
+  const handleMaxAgeUnitChange = React.useCallback(
+    (evt) => {
+      setMaxAgeUnits(Number(evt));
+    },
+    [setMaxAgeUnits]
+  );
+
+  const handleMaxSizeChange = React.useCallback(
+    (evt) => {
+      setMaxSize(Number(evt));
+    },
+    [setMaxSize]
+  );
+
+  const handleMaxSizeUnitChange = React.useCallback(
+    (evt) => {
+      setMaxSizeUnits(Number(evt));
+    },
+    [setMaxSizeUnits]
+  );
+
+  const handleTemplateChange = React.useCallback(
+    (templateName?: string, templateType?: TemplateType) => {
+      setTemplateName(templateName);
+      setTemplateType(templateType);
+    },
+    [setTemplateName, setTemplateType]
+  );
+
+  const getEventString = React.useCallback(() => {
+    var str = '';
+    if (!!templateName) {
+      str += `template=${templateName}`;
+    }
+    if (!!templateType) {
+      str += `,type=${templateType}`;
+    }
+    return str;
+  }, [templateName, templateType]);
+
+  const handleSubmit = React.useCallback(() => {
+    setIsLoading(true);
+    const options: RecordingOptions = {
+      toDisk: true,
+      maxAge: maxAge * maxAgeUnits,
+      maxSize: maxSize * maxSizeUnits,
+    };
+    const recordingAttributes: RecordingAttributes = {
+      name: automatedAnalysisRecordingName,
+      events: getEventString(),
+      duration: undefined,
+      archiveOnStop: false,
+      options: options,
+      metadata: {
+        labels: {
+          origin: automatedAnalysisRecordingName,
+        },
+      },
+    };
+    addSubscription(
+      context.api
+        .createRecording(recordingAttributes)
+        .pipe(first())
+        .subscribe({
+          next: (resp) => {
+            if (resp.ok && props.onCreate) {
+              props.onCreate();
+            }
+            setIsLoading(false);
+          },
+          error: (err) => {
+            setIsLoading(false);
+          },
+        })
+    );
+  }, [
+    addSubscription,
+    context.api,
+    setIsLoading,
+    getEventString,
+    props.onCreate,
+    maxAge,
+    maxAgeUnits,
+    maxSize,
+    maxSizeUnits,
+  ]);
+
+  const handleSaveConfig = React.useCallback(() => {
+    const options: AutomatedAnalysisRecordingConfig = {
+      templates: getEventString(),
+      maxAge: maxAge * maxAgeUnits,
+      maxSize: maxSize * maxSizeUnits,
+    };
+    context.settings.setAutomatedAnalysisRecordingConfig(options);
+    setIsSaveLoading(true);
+    const timer = setTimeout(() => {
```

Comment:
Here what we can do is for `context.settings.setAutomatedAnalysisRecordingConfig(options)` to return an observable that we can subscribe too and reset `isSaveLoading`, call `props.onSave` and `setShowHelperMessage`

Reply:
Oh, I was just adding the timeout cause it looks cooler. :-) I don't actually need it, I was just thinking users might want to think that the button is actually doing something. Thoughts?

Reply:
Yeh its looking cool to me too :D Ahh if so, I guess we can reduce it to around 300-500ms is enough :D It looks nice and fast :D